rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/jvminterop/JavaSystemInterface.kt

```diff
@@ -50,8 +52,23 @@ open class JavaSystemInterface(
     }
 
     // For calls from Java programs
-    private constructor (os: PrintStream, rpgSystem: RpgSystem) : this(os, rpgSystem::getProgram, { copyId -> rpgSystem.getCopy(copyId) }, rpgSystem = rpgSystem)
-    constructor (os: PrintStream) : this(os, RpgSystem())
+    private constructor (os: PrintStream, rpgSystem: RpgSystem, configuration: Configuration?) : this(os, rpgSystem::getProgram, { copyId -> rpgSystem.getCopy(copyId) }, rpgSystem = rpgSystem, configuration = configuration)
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem]
+     * */
+    constructor (os: PrintStream, configuration: Configuration?) : this(os, RpgSystem(), configuration)
+    /**
+     * Creates an instance of JavaSystemInterface with default [RpgSystem]
```

Comment:
From this comment, I don't understand the difference in logic from the previous instruction. Could you explain it to me?

Reply:
JavaSystemInterface has differents constructors (how to create an instance of itself) this constructor allows to create an instance of JavaSystemInterface by passing just two parameters because the instance of RpgSystem is automatically created.